### PR TITLE
show Controls characters in RTL direction

### PR DESCRIPTION
**Pull request by zayedalsaidi**

### CODE REVIEW

File: scribus/text/screenpainter.cpp [hunk 1]
```diff
@@ -125,6 +126,8 @@ void ScreenPainter::drawGlyph(const GlyphCluster& gc)
 		{
 			outline = m_item->doc()->symTab.copy();
 			chma4.translate(gc.width() - fontSize() * 0.7, -fontSize() * gc.scaleV() * 0.5);
+            if(gc.hasFlag(ScLayout_RightToLeft))
+                    flipHorizontal(chma4);
```
Comment (Fahad-Alsaidi) [inline scribus/text/screenpainter.cpp]: Please be sure to respect the indentation. The easiest way to do it in Qtcreator select the 4 lines and press ctrl+i

File: scribus/text/screenpainter.cpp [hunk 2]
```diff
@@ -140,11 +143,15 @@ void ScreenPainter::drawGlyph(const GlyphCluster& gc)
 		{
 			outline = m_item->doc()->symReturn.copy();
 			chma4.translate(gc.xoffset, -fontSize() * gc.scaleV() * 0.8);
+            if(gc.hasFlag(ScLayout_RightToLeft))
+                    flipHorizontal(chma4);
```
Comment (Fahad-Alsaidi) [inline scribus/text/screenpainter.cpp]: same here.

File: scribus/text/screenpainter.cpp [hunk 3]
```diff
@@ -140,11 +143,15 @@ void ScreenPainter::drawGlyph(const GlyphCluster& gc)
...
 		}
 		else if (gid == SpecialChars::LINEBREAK.unicode())
 		{
 			outline = m_item->doc()->symNewLine.copy();
 			chma4.translate(gc.xoffset, -fontSize() * gc.scaleV() * 0.4);
+            if(gc.hasFlag(ScLayout_RightToLeft))
+                    flipHorizontal(chma4);
```
Comment (Fahad-Alsaidi) [inline scribus/text/screenpainter.cpp]: same here.

File: scribus/text/screenpainter.cpp [hunk 4]
```diff
@@ -179,15 +186,15 @@ void ScreenPainter::drawGlyph(const GlyphCluster& gc)
 			outline.addQuadPoint(1, -10, 1, -10, 0, -10, 0, -10);
 		}
 		chma.scale(gc.scaleH() * fontSize() / 10.0, gc.scaleV() * fontSize() / 10.0);
-		outline.map(chma * chma4);
+        outline.map(chma*chma4);
```
Comment (Fahad-Alsaidi) [inline scribus/text/screenpainter.cpp]: same above plus put space before and after *

File: scribus/text/screenpainter.cpp [hunk 5]
```diff
@@ -179,15 +186,15 @@ void ScreenPainter::drawGlyph(const GlyphCluster& gc)
...
 		m_painter->setupPolygon(&outline, true);
 		QColor oldBrush = m_painter->brush();
 		m_painter->setBrush(gc.hasFlag(ScLayout_SuppressSpace) ? Qt::green
 															   : PrefsManager::instance()->appPrefs.displayPrefs.controlCharColor);
 		m_painter->setBrush(PrefsManager::instance()->appPrefs.displayPrefs.controlCharColor);
 		if (stroke)
 		{
-			QColor tmp = m_painter->pen();
+            QColor tmp = m_painter->pen();
```
Comment (Fahad-Alsaidi) [inline scribus/text/screenpainter.cpp]: same above, as general we use 4 tabs for indentation, please adjust the code style as screenshot below.
editor

**Comment by zayedalsaidi** [comment]
updated (I guess :) ). It seem I need more training on the workflow from Qt Creator to GitHub.

**Pull request merged**